### PR TITLE
BUILD: use system python3 in the chroot

### PR DESCRIPTION
**Pull request by mattip**
The chroot travis test used python2. Update it to use python3.

### CODE REVIEW

File: tools/travis-test.sh [hunk 1]
```diff
@@ -101,8 +101,8 @@ setup_chroot()
 
   # install needed packages
   sudo chroot $DIR bash -c "apt-get install -qq -y \
-    libatlas-base-dev gfortran python-dev python-nose python-pip cython \
-    python-pytest"
+    libatlas-base-dev gfortran python-dev python3-dev python-pip python3-pip \
+    cython  python3-pytest python-pytest"
```
Comment (charris) [inline tools/travis-test.sh]: Do we still need all the non-python3 versions if we are running Python3?

File: tools/travis-test.sh [hunk 2]
```diff
@@ -229,9 +229,9 @@ elif [ -n "$USE_CHROOT" ] && [ $# -eq 0 ]; then
   # the chroot'ed environment will not have the current locale,
   # avoid any warnings which may disturb testing
   export LANG=C LC_ALL=C
-  # run again in chroot with this time testing
+  # run again in chroot with this time testing with python3
   sudo linux32 chroot $DIR bash -c \
-    "cd numpy && PYTHON=python PIP=pip IN_CHROOT=1 $0 test"
+    "cd numpy && PYTHON=$PYTHON PIP=$PIP IN_CHROOT=1 $0 test"
```
Comment (charris) [inline tools/travis-test.sh]: Does `PIP=$PIP` make any sense here, it is already in the environment.

**Comment by charris** [comment]
This seems over complicated for its stated purpose.

File: tools/travis-test.sh (hunk 1 above)
Comment (mattip) [reply tools/travis-test.sh]: possibly, but then we cannot run chroot python2 tests anymore. If that is the direction I will change this

**Comment by mattip** [comment]
There are four changes:
1. change the test matrix to define the python and pip explicitly
1. percolate those choices into the chroot
1. install into the chroot python3 packages that were missing
1. replace `pip` with `$PIP` 

If we wish to only support python3 it can be simplified by removing 1, hardcoding python3 and pip3 into 2, and dropping the python2 packages from 3. 

I think 4 works outside the chroot because it is using a virtualenv, not clear to me if inside the chroot it is also in a virtualenv.

**Comment by charris** [comment]
I wonder if we can run 32 bit linux tests on azure-pipeline? We still have some slots before we hit the 10 parallel build limit. That doesn't help for releases yet, as they are still built on appveyor and travis, but it tmight be an easy way to get 32 testing on linux.

As it is running in bionic, the default python3 is 3.6, which should be good for a while. Did you try just changing things to `*3` in the chroot? And then in the calling lines
```
  sudo linux32 chroot $DIR bash -c \
    "cd numpy && PYTHON=python PIP=pip IN_CHROOT=1 $0 test"
```
use `python3` and `pip3`?

**Comment by charris** [comment]
Also, might be worth simplifying the i386 tests as suggested in the note
```
  # this can all be replaced with:
  # apt-get install libpython2.7-dev:i386
  # CC="gcc -m32" LDSHARED="gcc -m32 -shared" LDFLAGS="-m32 -shared" \
  #   linux32 python setup.py build
  # when travis updates to ubuntu 14.04
```

**Comment by mattip** [comment]
> Did you try just changing things to *3 in the chroot?

@charris Then we can only run chroot python3 tests. Is that OK, or do we wish to preserve the possibility of running python2 by changing `.travis.yml` only?

**Comment by mattip** [comment]
> this can all be replaced with ...

There are two parts to the `travis_test.sh` file: building and testing. That section may build a 32-bit numpy, if it links correctly with OpenBlas, but then we will need further changes to run the 32 bit python for testing.

**Comment by mattip** [comment]
> This seems over complicated for its stated purpose.

Do you mean my changes or the whole chroot 32 bit testing option in `travis-test.sh`? 

Travis itself suggests [using docker](https://github.com/travis-ci/travis-ci/issues/5770), here is one example of a [32 bit docker command](https://gitlab.com/zj/git/blob/master/ci/run-linux32-build.sh), it does not look less complicated than what we now have.

I could try pursuing the `linux32` option if so desired

**Comment by charris** [comment]
>  or do we wish to preserve the possibility of running python2.

We will be dropping Python2 support in a couple of weeks.

**Comment by mattip** [comment]
changed to only use python3 in chroot

**Comment by charris** [comment]
Thanks Matti.

**Comment by tylerjereddy** [comment]
I wonder if i.e., 32-bit build / test on a 64-bit OS without linalg acceleration libraries is prone to issues -- I think that's my current config there. My first pass at getting the `:i386` linear algebra stuff working was a bit of a mess, but the library otherwise builds and runs the first 30 % of the unit tests in full mode before stalling.

**Comment by charris** [comment]
Interesting. Does it always stall at the same test?

**Comment by tylerjereddy** [comment]
I think so, and same stall for 2.7 & 3.5. `libpython3.6-dev:i386` doesn't exist on the Azure Ubuntu version, but I think I can perhaps try `libc6-dev-i386` + 3.6-7 now that I'm mimicking the numba approach with 32-bit conda env + `gcc_linux-32 gxx_linux-32`

**Pull request merged**